src/module.js

```diff
@@ -1349,6 +1355,14 @@ export default class DailyIframe extends EventEmitter {
           this._iframe.src = '';
         }
         this._meetingState = DAILY_STATE_ERROR;
+        if (this._loadedCallback) {
+          this._loadedCallback(msg);
+          this._loadedCallback = null;
+        }
+        if (this._joinedCallback) {
+          this._joinedCallback(null, msg);
```

Comment:
oh wait one thing: `msg` here is an object of the form `{action: ..., errorMsg: ...}` right? we may want to resolve promises with just the error message itself and not the object.

Reply:
true. I guess even if `errorMsg` was `null/undefined/''` we're checking the `_meetingState` in the callbacks, so the promise would still reject. 👍